maven-plugin/plugin-core/src/main/java/org/jvnet/jaxb/maven/RawXJC2Mojo.java

```diff
@@ -577,6 +580,33 @@ private void addIfExistsToEpisodeSchemaBindings() throws MojoExecutionException
 		}
 	}
 
+    private void deleteEpisodeIfOnlyFileOnEmptyContext() {
+        if (!getEpisode() || getBuildContext() == null) {
+            return;
+        }
+        final File episodeFile = getEpisodeFile();
+        if (getBuildContext().isIncremental() && getBuildContext().newScanner(getGenerateDirectory()) instanceof EmptyScanner) {
+            if (!episodeFile.canWrite()) {
+                getLog().warn(MessageFormat
+                    .format("Episode file [{0}] is not writable, could not delete on incremental empty build.", episodeFile));
+                return;
+            }
+            DirectoryScanner directoryScanner = new DirectoryScanner();
+            directoryScanner.setBasedir(getGenerateDirectory());
+            directoryScanner.setIncludes(getProduces());
+            directoryScanner.scan();
+
+            if (directoryScanner.getIncludedFiles() != null && directoryScanner.getIncludedFiles().length == 1) {
+                String fileFound = directoryScanner.getIncludedFiles()[0];
+                if (new File(directoryScanner.getBasedir(), fileFound).equals(episodeFile)) {
+                    getLog().info(MessageFormat
+                        .format("directoryScanner got only one file [{0}] which is episodeFile - deleting since running in m2e context.", fileFound, episodeFile));
```

Comment:
Not sure we should log 'm2e' if there is a non-Eclipse way for this scenario to occur.

Reply:
Fixed